Exclude hidden links from get all links

ivshapovalov opened the pull request:
close #62  or issue #49

ivshapovalov committed:
#49 Fix error in CORS Filter

ivshapovalov committed:
#49 Add 'hidden' field in Link entity

ivshapovalov committed:
#49 Update repository 'getAllLinks' logic with 'hidden' field

ivshapovalov committed:
#49 Update Repository Unit Tests (increase coverage)

ivshapovalov committed:
#49 Add ApiModel to SaveLinkRequest

ivshapovalov committed:
#49 Some refactor

Benkoff commented on links/src/main/java/juja/microservices/links/repository/impl/LinksRepositoryImpl.java:
```diff
@@ -51,6 +51,10 @@ public Link getLinkByURL(String url) throws NotFoundException {
+        query.addCriteria(Criteria.where("hidden").is(false));
+        List<Link> result = mongoTemplate.find(query, Link.class, mongoCollectionName);
+        log.info("Found {} active links in database", result);
+        return result;
```
i doubt that this is a good idea to apply filters to getAllLinks() method at repository level. 
what about a new one like getAllNotHiddenLinks or anything similar?

ivshapovalov replied:
Exactly. Thanks

Benkoff commented on links/src/main/java/juja/microservices/links/model/Link.java:
```diff
@@ -1,30 +1,31 @@
+    @ApiModelProperty(value = "URL of saved link", required = true)
     private String url;
 
+    @ApiModelProperty(value = "Flag that means link is hidden or active")
```
i'd remove final part "or active". if you want to use it as a substitute for "not hidden", there are a lot of more evident words like open, public, visible, apparent so on. but as for me its better to use a simple negation: "Flag that means the link is hidden or **not**".

ivshapovalov replied:
You're right

ivshapovalov committed:
#49 Rename methods appropriately

ivshapovalov committed:
Merge remote-tracking branch 'upstream/develop' into 49-exclude-hidden-links-from-get-all-links

# Conflicts:
#	links/src/main/java/juja/microservices/links/config/CorsConfig.java
#	links/src/main/java/juja/microservices/links/controller/LinksController.java
#	links/src/main/java/juja/microservices/links/model/Link.java
#	links/src/main/java/juja/microservices/links/model/SaveLinkRequest.java
#	links/src/main/java/juja/microservices/links/repository/impl/LinksRepositoryImpl.java
#	links/src/test/java/juja/microservices/links/controller/LinksControllerTest.java
#	links/src/test/java/juja/microservices/links/service/LinksServiceTest.java

ivshapovalov committed:
#49 Update Swagger to 2.8.0

ivshapovalov committed:
#49 Create builder in entities

ivshapovalov committed:
#49 Use Entities builder in code and tests

ivshapovalov committed:
#49 Some refactoring

ivshapovalov committed:
#49 Remove builder from entities

ivshapovalov committed:
#49 Replace criteria in Repository getAllNotHiddenLinks() method
    "hidden".ne(true) obtain links without "hidden" fields
    we suppose that links without "hidden" field are equals
    links with hidden default value 'false'
